cmd/scan_txs.cpp

```diff
@@ -0,0 +1,94 @@
+/*
+   Copyright 2020 The Silkworm Authors
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+#include <absl/container/flat_hash_set.h>
+#include <absl/flags/flag.h>
+#include <absl/flags/parse.h>
+#include <absl/flags/usage.h>
+#include <absl/time/time.h>
+
+#include <boost/filesystem.hpp>
+#include <iostream>
+#include <silkworm/db/access_layer.hpp>
+#include <silkworm/execution/execution.hpp>
+
+ABSL_FLAG(std::string, datadir, silkworm::db::default_path(), "chain DB path");
+ABSL_FLAG(uint64_t, from, 1, "start from block number (inclusive)");
+ABSL_FLAG(uint64_t, to, UINT64_MAX, "check up to block number (exclusive)");
+
+int main(int argc, char* argv[]) {
+    absl::SetProgramUsageMessage("Executes Ethereum blocks and compares resulting change sets against DB.");
+    absl::ParseCommandLine(argc, argv);
+
+    namespace fs = boost::filesystem;
+
+    fs::path db_path(absl::GetFlag(FLAGS_datadir));
+    if (!fs::exists(db_path) || !fs::is_directory(db_path) || db_path.empty()) {
+        std::cerr << absl::GetFlag(FLAGS_datadir) << " does not exist.\n";
+        std::cerr << "Use --db flag to point to a Turbo-Geth populated chaindata.\n";
+        return -1;
+    }
+
+    using namespace silkworm;
+
+    lmdb::options db_opts{};
+    db_opts.read_only = true;
+    std::shared_ptr<lmdb::Environment> env{lmdb::get_env(absl::GetFlag(FLAGS_datadir).c_str(), db_opts)};
```

Comment:
This statement might throw. I'd suggest to enclose the whole loop in a try catch block to avoid unhandled errors

Reply:
I'll add that and repost.